File: backend/src/main/java/com/fullteaching/backend/model/CourseInvitationNotification.java
package com.fullteaching.backend.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class CourseInvitationNotification extends Notification {
    public CourseInvitationNotification(String message, User user, Course course) {
        super(message, user, course, NotificationType.INVITED_TO_COURSE);
    }
}

File: frontend/src/app/classes/course-invitation-notification.ts
import {Notification} from './notification';
import {User} from './user';
import {Course} from './course';

export class CourseInvitationNotification extends Notification {
  constructor(id: number, message: string, user: User, course: Course) {
    super(id, message, user, course, `New course invitation!`);
    this.course = course;
  }
}
